design/one-pager-e2e-tests.md

```diff
@@ -0,0 +1,170 @@
+
+* Owner: Nic Cope (@negz)
+* Reviewers: Hasan Turken (@turkenh)
+* Status: Accepted
+
+## Background
+
+Crossplane relies heavily on unit tests to ensure quality, but these alone are
+not enough to ensure Crossplane functions correctly as a whole.
+
+Thus far we have relied on an honor system of folks running manual black-box
+tests to increase the likelihood that their changes work in real-world
+conditions. These tests:
+
+* Are not very repeatable, or portable for others to run.
+* Are often only run on one environment - i.e. `kind` on the author's laptop.
+* Are not re-run on future changes to catch regressions.
+* Are likely to miss things like performance degradation.
+* Are usually skipped for innocuous-seeming things like dependency bumps.
+* Often don't consider interactions with features besides the one under test.
+* Often don't cover upgrade scenarios.
+
+Over the years this has lead to several (probably many) issues in released
+versions of Crossplane that would _likely_ have been caught by E2E tests. Some
+examples include:
+
+* https://github.com/crossplane/crossplane/pull/3376
+* https://github.com/crossplane/crossplane/issues/3113
+* https://github.com/crossplane/crossplane/issues/3071
+
+We've made several attempts to add integration and E2E tests to Crossplane in
+the past, but none have flourished. Typically we've aligned on a pattern,
+implemented the first test or two, then never touched them again. I believe this
+is largely because we've relied on CNCF interns to kick-off the testing effort. 
+The testing frameworks we've put into place are actually promising - it's just
+hard to ensure contributors keep using them once your internship ends.
+
+Examples include:
+
+* https://github.com/crossplane/crossplane/blob/v1.12.2/cluster/local/integration_tests.sh
+* https://github.com/crossplane/crossplane/tree/v1.12.2/test/e2e
+* https://github.com/crossplane/test
+
+`integration_tests.sh` is our most long-lived attempt at black-box testing. It
+runs as part of every continuous integration (CI) job (i.e. on every PR) and
+simply ensures Crossplane starts without crashing.
+
+`crossplane/test` does run on a regular schedule, using a mix of tests defined
+under `test/e2e` in c/c, and tests defined in its own repository. It covers:
+
+* Upgrading Crossplane from stable to master builds, using the Helm chart.
+* Upgrading a Provider from one version to another.
+* Resolving Configuration package dependencies.
+* Creating a claim using a very minimal Composition, without any patches or
+  transforms.
+
+Because it's run in a separate repository and isn't connected to PRs or releases
+it's unclear whether anyone notices if/when tests break. It's also easy to
+forget to update the tests when adding new functionality.
+
+## Goals
+
+The goal of this proposal is to:
+
+* Automatically simulate real-world use of all of Crossplane's functionality.
+* Catch the kind of bugs that unit tests miss, before they're released.
```

Comment:
Definitely - will update.

I do think eventually it might be tough to run the entire suite on every PR, but we can address that when we get to it. (Maybe add a `/test` trigger similar to upjet, Kube, etc).